### ISSUE TITLE
Pick List, Pick List Group and Pick List Item - Ensure custom controls provide proper textual name, role, and state information - (2036570119)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5595]
### Module: 
 24 Pick List, Pick List Group and Pick List Item 

### Violation: 
 Ensure custom controls provide proper textual name, role, and state information 

![image](https://user-images.githubusercontent.com/5023024/202743537-5a00c69d-445f-4431-a03e-82c9163db39c.png)

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570119 

### Severity: 
 7 

### Description: 
 [Issue]
There are checkboxes without appropriate role, and or state information. Examples include:
- Select multiple items from list

[User Impact]
Screen reader users will be unable to determine that these controls can be checked or unchecked. They will also be unable to review which options they have chosen.

[Code Reference] 
 ```
<calcite-pick-list-item label="River" description="A large natural stream of water flowing in a channel to the sea, a lake, or another such stream." value="river" icon="square" intl-remove="Remove" calcite-hydrated="" selected="">
        <calcite-action text="Enable River layer" slot="actions-end" icon="layer" appearance="solid" scale="m" calcite-hydrated=""></calcite-action>
      </calcite-pick-list-item>
 ``` 

### Note: 
 [Suggestion]
Ensure custom controls provide proper textual name, role, and state information.

The best way to do this is to use native controls, as they come with this information built in. In this case, ``` <input type="checkbox"> ``` elements would be most appropriate.

Developers can often achieve the same visual appearance using progressively enhanced CSS. Alternatively, they can visually hide the checkboxes with opacity: 0 and use SVGs to fake their appearance.

If conversion to native checkboxes is not possible, the following information is expected:
- The element functioning as the checkbox must have role="checkbox".
- This checkbox element must have a label provided by internal text, aria-label, or aria-labelledby.
- When checked, the checkbox element must have aria-checked="true", and when not checked, it must have aria-checked="false". If partially checked, it must have aria-checked="mixed".
- Checkboxes in a logical group must be children of an element with role="group". That element must have aria-labelledby set to the ID of text labelling the group.

[Recommended Reading]
For more information about creating accessible comboboxes, please see the W3C Web Accessibility Initiative&#039;s ARIA Authoring Practices Guide: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/ 

### Media Type: 
 Custom Controls 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by driskull** [comment]
This can be configured by the end user.
